### PR TITLE
NotFoundException -> Exceptions\ServiceNotFound

### LINKED ISSUE #2: Align this repository style-wise with other packages in the `technically` namespace
- [ ] Move exception class to `Exceptions` sub-namespace
- [ ] Rename exception class to `ServiceNotFound`

See [technically/null-container](https://github.com/technically-php/null-container/blob/main/src/Exceptions/ServiceNotFound.php) as example.

### PR DESCRIPTION
**Pull request by someniatko**
closes #2

### CODE REVIEW

File: src/Exception/ServiceNotFound.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Technically\WhitelistContainer\Exception;
```
Comment (e1himself) [inline src/Exception/ServiceNotFound.php:5]: It should be plural: `Exceptions`.

> Move exception class to `Exceptions` sub-namespace

File: src/Exception/ServiceNotFound.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+use Psr\Container\NotFoundExceptionInterface;
+
+/** @internal */
```
Comment (e1himself) [inline src/Exception/ServiceNotFound.php:9]: Convert this to mutliline doc-block please.

File: src/WhitelistContainer.php [hunk 3]
```diff
@@ -31,7 +32,7 @@ public function __construct(
     final public function get(string $id)
     {
         if (! $this->has($id)) {
-            throw new NotFoundException($id . ' not found in "' . $this->containerName . '" container');
+            throw new Exception\ServiceNotFound($id . ' not found in "' . $this->containerName . '" container');
```
Comment (e1himself) [inline src/WhitelistContainer.php:35]: Why `Exception\ServiceNotFound`?

**Review by e1himself — CHANGES_REQUESTED** [review]
Hi @someniatko!

Thanks for submitting this PR :+1: 

I have a couple of comments about the code. Please see below.

File: src/WhitelistContainer.php (hunk 3 above)
Comment (someniatko) [reply src/WhitelistContainer.php:35]: @e1himself dunno, IMO it feels more obvious that it's an exception, as most devs are used to the Exception suffix.

**Comment by someniatko** [comment]
@e1himself It seems I can no longer push to this project / branch. Did you change anything regarding rights?

**Comment by e1himself** [comment]
> It seems I can no longer push to this project / branch. Did you change anything regarding rights?

@someniatko Sorry, I've been checking some settings in the organisation and it probably affected your permissions. Can you please try now?

File: src/WhitelistContainer.php (hunk 3 above)
Comment (e1himself) [reply src/WhitelistContainer.php]: I understand your desire seeing `Exception` next an exception class name. But isn't it clear-enough that anything used with `throw` is an exception? 

You mind find interesting reading this article: https://mnapoli.fr/approaching-coding-style-rationally/, specifically the last part about "Exception" suffix.

**Comment by someniatko** [comment]
@e1himself LGTY?

**Review by e1himself — APPROVED** [review]
> LGTY?

Yes!



PS. JFYI there's a nice little button to re-request a review :smiley: 
Знімок з 2021-06-25 14-54-40

File: src/WhitelistContainer.php (hunk 3 above)
Comment (e1himself) [reply src/WhitelistContainer.php]: @someniatko I have an impression you might have missed my comment above, as you didn't react to this thread in any way. Did you see it? What do you think?

**Pull request merged**